python/paddle/incubate/optimizer/functional/lbfgs_v2.py

```diff
@@ -0,0 +1,412 @@
+        history_size (int, optional): update history size. The default value is 100.
+        line_search_fn (string, optional): either 'strong_wolfe' or None. The default value is strong_wolfe.
+        parameters (list|tuple, optional): List/Tuple of ``Tensor`` names to update to minimize ``loss``. \
+            This parameter is required in dygraph mode. And you can specify different options for \
+            different parameter groups such as the learning rate, weight decay, etc, \
+            then the parameters are list of dict. Note that the learning_rate in paramter groups \
+            represents the scale of base learning_rate. \
+            The default value is None in static mode, at this time all parameters will be updated.
+        weight_decay (float|WeightDecayRegularizer, optional): The strategy of regularization. \
+            It canbe a float value as coeff of L2 regularization or \
+            :ref:`api_fluid_regularizer_L1Decay`, :ref:`api_fluid_regularizer_L2Decay`.
+            If a parameter has set regularizer using :ref:`api_fluid_ParamAttr` already, \
+            the regularization setting here in optimizer will be ignored for this parameter. \
+            Otherwise, the regularization setting here in optimizer will take effect. \
+            Default None, meaning there is no regularization.
+        grad_clip (GradientClipBase, optional): Gradient cliping strategy, it's an instance of \
+            some derived class of ``GradientClipBase`` . There are three cliping strategies \
+            ( :ref:`api_fluid_clip_GradientClipByGlobalNorm` , :ref:`api_fluid_clip_GradientClipByNorm` , \
+            :ref:`api_fluid_clip_GradientClipByValue` ). Default None, meaning there is no gradient clipping.
+        name (str, optional): Normally there is no need for user to set this property.
+            For more information, please refer to :ref:`api_guide_Name`.
+            The default value is None.
+
+    Return:
+        loss: the final loss of closure
+
+    Examples:
+    .. code-block:: python
+
+    import paddle
+    import numpy as np
+    from paddle.incubate.optimizer.functional import LBFGS
+
+    np.random.seed(0)
+    np_w = np.random.rand(1).astype(np.float32)
+    np_x = np.random.rand(1).astype(np.float32)
+
+    inputs = [np.random.rand(1).astype(np.float32) for i in range(10)]
+    # y = 2x
+    targets = [2 * x for x in inputs]
+
+    class Net(paddle.nn.Layer):
+        def __init__(self):
+            super(Net, self).__init__()
+            w = paddle.to_tensor(np_w)
+            self.w = paddle.create_parameter(shape=w.shape, dtype=w.dtype, default_initializer=paddle.nn.initializer.Assign(w))
+
+        def forward(self, x):
+            return self.w * x
+
+    net = Net()
+    opt = LBFGS(lr=1, max_iter=1, max_eval=None, tolerance_grad=1e-07, tolerance_change=1e-09, history_size=100, line_search_fn='strong_wolfe', parameters=net.parameters())
+    def train_step(inputs, targets):
```

Comment:
No need to define this train step

Reply:
it‘s convenient for the different input in different batch